<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Survival Analysis with R</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />
<!-- Font Awesome -->
<!-- <link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" /> -->
<!-- <script src="https://use.fontawesome.com/54ee8c2dfd.js"></script> -->

<!-- Google fonts -->
<link href='https://fonts.googleapis.com/css?family=Open+Sans:400,400italic,700,700italic|Oswald:400,700' rel='stylesheet' type='text/css'>

<!-- Favicon -->
<link rel="shortcut icon" type="image/x-icon" href="img/favicon.ico">

<!-- Google analytics -->
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');
  ga('create', 'UA-8298649-8', 'auto');
  ga('send', 'pageview');
</script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/font-awesome/4.5.0/css/font-awesome.min.css" type="text/css" />
<link rel="stylesheet" href="styles.css" type="text/css" />

</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
}

</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">BIMS8382</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li>
  <a href="setup.html">
    <span class="fa fa-cog"></span>
     
    Setup
  </a>
</li>
<li>
  <a href="data.html">
    <span class="fa fa-download"></span>
     
    Data
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-book"></span>
     
    Workshops
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="r-basics.html">R Basics</a>
    </li>
    <li>
      <a href="r-dataframes.html">Data Frames</a>
    </li>
    <li>
      <a href="r-dplyr-yeast.html">Data Manipulation</a>
    </li>
    <li>
      <a href="r-tidy.html">Tidying data</a>
    </li>
    <li>
      <a href="r-lists.html">List Manipulation</a>
    </li>
    <li>
      <a href="r-interactive-viz.html">Interactive Visualization with JavaScript and R</a>
    </li>
    <li>
      <a href="r-shiny.html">Building Shiny Web Apps in R</a>
    </li>
    <li>
      <a href="r-ncbi.html">Harvesting Data from NCBI</a>
    </li>
    <li>
      <a href="r-viz-gapminder.html">Data Visualization</a>
    </li>
    <li>
      <a href="r-rmarkdown.html">Reproducible Research &amp; RMarkdown</a>
    </li>
    <li>
      <a href="r-stats.html">Essential Statistics with R</a>
    </li>
    <li>
      <a href="r-survival.html">Survival Analysis with TCGA Data</a>
    </li>
    <li>
      <a href="r-ggtree.html">Drawing &amp; annotating phylogenetic trees</a>
    </li>
    <li>
      <a href="r-rnaseq-airway.html">RNA-seq Data Analysis</a>
    </li>
    <li>
      <a href="r-predictive-modeling.html">Predictive Analytics &amp; Forecasting Influenza</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    BIMS8382
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="bims8382.html">
        <span class="fa fa-graduation-cap"></span>
         
        Syllabus
      </a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">--- Lessons ---</li>
    <li>
      <a href="r-basics.html">R Basics</a>
    </li>
    <li>
      <a href="r-dataframes.html">Data Frames</a>
    </li>
    <li>
      <a href="r-dplyr-yeast.html">Data Manipulation</a>
    </li>
    <li>
      <a href="r-tidy.html">Tidying data</a>
    </li>
    <li>
      <a href="r-viz-gapminder.html">Data Visualization</a>
    </li>
    <li>
      <a href="r-rmarkdown.html">Reproducible Research &amp; RMarkdown</a>
    </li>
    <li>
      <a href="r-stats.html">Essential Statistics with R</a>
    </li>
    <li>
      <a href="r-survival.html">Survival Analysis with TCGA Data</a>
    </li>
    <li>
      <a href="r-rnaseq-airway.html">RNA-Seq Data Analysis</a>
    </li>
    <li class="divider"></li>
    <li class="dropdown-header">--- Homework ---</li>
    <li>
      <a href="r-dplyr-homework.html">Data Manipulation</a>
    </li>
    <li>
      <a href="r-viz-homework.html">Data Visualization</a>
    </li>
    <li>
      <a href="r-rmarkdown-homework.html">Reproducible Research &amp; RMarkdown</a>
    </li>
    <li>
      <a href="r-stats-homework.html">Essential Statistics</a>
    </li>
    <li>
      <a href="r-rnaseq-homework.html">RNA-seq Data Analysis</a>
    </li>
    <li>
      <a href="r-predictive-modeling.html">Predictive Analytics &amp; Forecasting Influenza</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    <span class="fa fa-question fa-lg"></span>
     
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="people.html">People</a>
    </li>
    <li>
      <a href="help.html">Further resources</a>
    </li>
    <li>
      <a href="https://github.com/bioconnector/workshops">Source code for this site</a>
    </li>
  </ul>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Survival Analysis with R</h1>

</div>


<p>This class will provide hands-on instruction and exercises covering survival analysis using R. Some of the data to be used here will come from The Cancer Genome Atlas (TCGA), where we may also cover programmatic access to TCGA through Bioconductor if time allows.</p>
<p><strong>Prerequisites: <a href="r-basics.html">Familiarity with R</a> is <em>required</em></strong> (including working with <a href="r-dataframes.html">data frames</a>, installing/using packages, importing data, and saving results); familiarity with <a href="r-dplyr-yeast.html">dplyr</a> and <a href="r-viz-gapminder.html">ggplot2</a> packages is highly recommended.</p>
<p><strong>You must <a href="setup.html#survival_analysis">complete the setup here</a> <em>prior to class</em>.</strong> This includes installing R, RStudio, and the required packages under the <a href="setup.html#survival_analysis">“Survival Analysis” heading</a>. Please <a href="people.html">contact one of the instructors</a> <em>prior to class</em> if you are having difficulty with any of the setup. Please bring your laptop and charger cable to class.</p>
<p><strong>Handouts</strong>: Download and print out these handouts and bring them to class:</p>
<ul>
<li><a href="handouts/r-survival-cheatsheet.pdf">Cheat sheet</a></li>
<li><a href="handouts/r-survival-handout.pdf">Background handout</a></li>
<li><a href="handouts/r-survival-exercises.pdf">Exercises handout</a></li>
</ul>
<!-- **Slides**: [click here](slides/r-survival.html). -->
<div id="background" class="section level2">
<h2>Background</h2>
<p>In the class on <a href="r-stats.html">essential statistics</a> we covered basic categorical data analysis – comparing proportions (risks, rates, etc) between different groups using a chi-square or fisher exact test, or logistic regression. For example, we looked at how the diabetes rate differed between males and females. In this kind of analysis you implicitly assume that the rates are constant over the period of the study, or as defined by the different groups you defined.</p>
<p>But, in longitudinal studies where you track samples or subjects from one time point (e.g., entry into a study, diagnosis, start of a treatment) until you observe some outcome <em>event</em> (e.g., death, onset of disease, relapse), it doesn’t make sense to assume the rates are constant. For example: the risk of death after heart surgery is highest immediately post-op, decreases as the patient recovers, then rises slowly again as the patient ages. Or, recurrence rate of different cancers varies highly over time, and depends on tumor genetics, treatment, and other environmental factors.</p>
<div id="definitions" class="section level3">
<h3>Definitions</h3>
<p><strong>Survival analysis</strong> lets you analyze the rates of occurrence of events over time, without assuming the rates are constant. Generally, survival analysis lets you model the <em>time until an event occurs</em>,<a href="#fn1" class="footnoteRef" id="fnref1"><sup>1</sup></a> or compare the time-to-event between different groups, or how time-to-event correlates with quantitative variables.</p>
<p>The <strong>hazard</strong> is the instantaneous event (death) rate at a particular time point <em>t</em>. Survival analysis doesn’t assume the hazard is constant over time. The <em>cumulative hazard</em> is the total hazard experienced up to time <em>t</em>.</p>
<p>The <strong>survival function</strong>, is the probability an individual survives (or, the probability that the event of interest does not occur) up to and including time <em>t</em>. It’s the probability that the event (e.g., death) hasn’t occured yet. It looks like this, where <span class="math inline">\(T\)</span> is the time of death, and <span class="math inline">\(Pr(T&gt;t)\)</span> is the probability that the time of death is greater than some time <span class="math inline">\(t\)</span>. <span class="math inline">\(S\)</span> is a probability, so <span class="math inline">\(0 \leq S(t) \leq 1\)</span>, since survival times are always positive (<span class="math inline">\(T \geq 0\)</span>).</p>
<p><span class="math display">\[ S(t) = Pr(T&gt;t) \]</span></p>
<p>The <strong>Kaplan-Meier</strong> curve illustrates the survival function. It’s a step function illustrating the cumulative survival probability over time. The curve is horizontal over periods where no event occurs, then drops vertically corresponding to a change in the survival function at each time an event occurs.</p>
<p><strong>Censoring</strong> is a type of missing data problem unique to survival analysis. This happens when you track the sample/subject through the end of the study and the event never occurs. This could also happen due to the sample/subject dropping out of the study for reasons other than death, or some other loss to followup. The sample is <em>censored</em> in that you only know that the individual survived up to the loss to followup, but you don’t know anything about survival after that.<a href="#fn2" class="footnoteRef" id="fnref2"><sup>2</sup></a></p>
<p><strong>Proportional hazards assumption</strong>: The main goal of survival analysis is to compare the survival functions in different groups, e.g., leukemia patients as compared to cancer-free controls. If you followed both groups until everyone died, both survival curves would end at 0%, but one group might have survived on average a lot longer than the other group. Survival analysis does this by comparing the <em>hazard</em> at different times over the observation period. Survival analysis doesn’t assume that the hazard is constant, but <em>does</em> assume that the <em>ratio</em> of hazards between groups is constant over time.<a href="#fn3" class="footnoteRef" id="fnref3"><sup>3</sup></a> This class does <em>not</em> cover methods to deal with non-proportional hazards, or interactions of covariates with the time to event.</p>
<p><strong>Proportional hazards regression</strong> a.k.a. <strong>Cox regression</strong> is the most common approach to assess the effect of different variables on survival.</p>
</div>
<div id="cox-ph-model" class="section level3">
<h3>Cox PH Model</h3>
<p>Kaplan-Meier curves are good for visualizing differences in survival between two categorical groups,<a href="#fn4" class="footnoteRef" id="fnref4"><sup>4</sup></a> but they don’t work well for assessing the effect of <em>quantitative</em> variables like age, gene expression, leukocyte count, etc. Cox PH regression can assess the effect of both categorical and continuous variables, and can model the effect of multiple variables at once.<a href="#fn5" class="footnoteRef" id="fnref5"><sup>5</sup></a></p>
<p>Cox PH regression models the natural log of the hazard at time <em>t</em>, denoted <span class="math inline">\(h(t)\)</span>, as a function of the baseline hazard (<span class="math inline">\(h_0(t)\)</span>) (the hazard for an individual where all exposure variables are 0) and multiple exposure variables <span class="math inline">\(x_1\)</span>, <span class="math inline">\(x_1\)</span>, <span class="math inline">\(...\)</span>, <span class="math inline">\(x_p\)</span>. The form of the Cox PH model is:</p>
<p><span class="math display">\[ log(h(t)) = log(h_0(t)) + \beta_1 x_1 + \beta_2 x_2 + ... + \beta_p x_p \]</span></p>
<p>If you exponentiate both sides of the equation, and limit the right hand side to just a single categorical exposure variable (<span class="math inline">\(x_1\)</span>) with two groups (<span class="math inline">\(x_1=1\)</span> for exposed and <span class="math inline">\(x_1=0\)</span> for unexposed), the equation becomes:</p>
<p><span class="math display">\[ h_1(t) = h_0(t) \times e^{\beta_1 x_1} \]</span></p>
<p>Rearranging that equation lets you estimate the <strong>hazard ratio</strong>, comparing the exposed to the unexposed individuals at time <em>t</em>:</p>
<p><span class="math display">\[ HR(t) = \frac{h_1(t)}{h_0(t)} = e^{\beta_1} \]</span></p>
<p>This model shows that <strong>the hazard ratio is <span class="math inline">\(e^{\beta_1}\)</span>,</strong> and remains constant over time <em>t</em> (hence the name <em>proportional hazards regression</em>). The <span class="math inline">\(\beta\)</span> values are the regression coefficients that are estimated from the model, and represent the <span class="math inline">\(log(Hazard\, Ratio)\)</span> for each unit increase in the corresponding predictor variable. The interpretation of the hazards ratio depends on the measurement scale of the predictor variable, but in simple terms, a positive coefficient indicates worse survival and a negative coefficient indicates better survival for the variable in question.</p>
</div>
</div>
<div id="survival-analysis-in-r" class="section level2">
<h2>Survival analysis in R</h2>
<p>The core survival analysis functions are in the <strong><a href="https://cran.r-project.org/web/packages/survival/">survival</a></strong> package. The survival package is one of the few “core” packages that comes bundled with your basic R installation, so you probably didn’t need to <code>install.packages()</code> it. But, you’ll need to load it like any other library when you want to use it. We’ll also be using the <strong>dplyr</strong> package, so let’s load that too. Finally, we’ll also want to load the <strong><a href="https://cran.rstudio.com/web/packages/survminer/index.html">survminer</a></strong> package, which provides much nicer Kaplan-Meier plots out-of-the-box than what you get out of base graphics.</p>
<pre class="r"><code>library(dplyr)
library(survival)
library(survminer)</code></pre>
<p>The core functions we’ll use out of the survival package include:</p>
<ul>
<li><code>Surv()</code>: Creates a survival object.</li>
<li><code>survfit()</code>: Fits a survival curve using either a formula, of from a previously fitted Cox model.</li>
<li><code>coxph()</code>: Fits a Cox proportional hazards regression model.</li>
</ul>
<p>Other optional functions you might use include:</p>
<ul>
<li><code>cox.zph()</code>: Tests the proportional hazards assumption of a Cox regression model.</li>
<li><code>survdiff()</code>: Tests for differences in survival between two groups using a log-rank / Mantel-Haenszel test.<a href="#fn6" class="footnoteRef" id="fnref6"><sup>6</sup></a></li>
</ul>
<p><code>Surv()</code> creates the response variable, and typical usage takes the time to event,<a href="#fn7" class="footnoteRef" id="fnref7"><sup>7</sup></a> and whether or not the event occured (i.e., death vs censored). <code>survfit()</code> creates a survival curve that you could then display or plot. <code>coxph()</code> implements the regression analysis, and models specified the same way as in regular linear models, but using the <code>coxph()</code> function.</p>
<div id="getting-started" class="section level3">
<h3>Getting started</h3>
<p>We’re going to be using the built-in lung cancer dataset<a href="#fn8" class="footnoteRef" id="fnref8"><sup>8</sup></a> that ships with the survival package. You can get some more information about the dataset by running <code>?lung</code>. The help tells us there are 10 variables in this data:</p>
<pre class="r"><code>library(survival)
?lung</code></pre>
<ol style="list-style-type: decimal">
<li><code>inst</code>: Institution code</li>
<li><strong><code>time</code>: Survival time in days</strong></li>
<li><strong><code>status</code>: censoring status 1=censored, 2=dead</strong></li>
<li><code>age</code>: Age in years</li>
<li><strong><code>sex</code>: Male=1 Female=2</strong></li>
<li><code>ph.ecog</code>: ECOG performance score (0=good 5=dead)</li>
<li><code>ph.karno</code>: Karnofsky performance score as rated by physician</li>
<li><code>pat.karno</code>: Karnofsky performance score as rated by patient</li>
<li><code>meal.cal</code>: Calories consumed at meals</li>
<li><code>wt.loss</code>: Weight loss in last six months</li>
</ol>
<p>You can access the data just by running <code>lung</code>, as if you had read in a dataset and called it <code>lung</code>. You can operate on it just like any other data frame.</p>
<pre class="r"><code>head(lung)
class(lung)
dim(lung)
View(lung)</code></pre>
<p>Notice that lung is a plain <code>data.frame</code> object. You could see what it looks like as a tibble (prints nicely, tells you the type of variable each column is). You could then reassign <code>lung</code> to the <code>as_tibble()</code>-ified version.</p>
<pre class="r"><code>as_tibble(lung)
lung &lt;- as_tibble(lung)
lung</code></pre>
</div>
<div id="survival-curves" class="section level3">
<h3>Survival Curves</h3>
<p>Check out the help for <code>?Surv</code>. This is the main function we’ll use to create the survival object. You can play fast and loose with how you specify the arguments to <code>Surv</code>. The help tells you that when there are two unnamed arguments, they will match <code>time</code> and <code>event</code> in that order. This is the common shorthand you’ll often see for right-censored data. The alternative lets you specify interval data, where you give it the start and end times (<code>time</code> and <code>time2</code>). If you keep reading you’ll see how <code>Surv</code> tries to guess how you’re coding the status variable. It will try to guess whether you’re using 0/1 or 1/2 to represent censored vs “dead”, respectively.<a href="#fn9" class="footnoteRef" id="fnref9"><sup>9</sup></a></p>
<p>Try creating a survival object called <code>s</code>, then display it. If you go back and <code>head(lung)</code> the data, you can see how these are related. It’s a special type of vector that tells you both how long the subject was tracked for, and whether or not the event occured or the sample was censored (shown by the <code>+</code>).</p>
<pre class="r"><code>s &lt;- Surv(lung$time, lung$status)
class(s)
s
head(lung)</code></pre>
<p>Now, let’s <strong>fit a survival curve</strong> with the <code>survfit()</code> function. See the help for <code>?survfit</code>. Here we’ll create a simple survival curve that doesn’t consider any different groupings, so we’ll specify just an intercept (e.g., <code>~1</code>) in the formula that <code>survfit</code> expects. We can do what we just did by “modeling” the survival object <code>s</code> we just created against an intercept only, but from here out, we’ll just do this in one step by nesting the <code>Surv()</code> call within the <code>survfit()</code> call, and similar to how we specify data for linear models with <code>lm()</code>, we’ll use the <code>data=</code> argument to specify which data we’re using. Similarly, we can assign that to another object called <code>sfit</code> (or whatever we wanted to call it).</p>
<pre class="r"><code>survfit(s~1)
survfit(Surv(time, status)~1, data=lung)
sfit &lt;- survfit(Surv(time, status)~1, data=lung)
sfit</code></pre>
<p>Now, that object itself isn’t very interesting. It’s more interesting to run <code>summary</code> on what it creates. This will show a life table.</p>
<pre class="r"><code>summary(sfit)</code></pre>
<p>These tables show a row for each time point where either the event occured or a sample was censored. It shows the number at risk (number still remaining), and the cumulative survival at that instant.</p>
<p>What’s more interesting though is if we model something besides just an intercept. Let’s fit survival curves separately by sex.</p>
<pre class="r"><code>sfit &lt;- survfit(Surv(time, status)~sex, data=lung)
sfit
summary(sfit)</code></pre>
<p>Now, check out the help for <code>?summary.survfit</code>. You can give the <code>summary()</code> function an option for what times you want to show in the results. Look at the range of followup times in the lung dataset with <code>range()</code>. You can create a sequence of numbers going from one number to another number by increments of yet another number with the <code>seq()</code> function.</p>
<pre class="r"><code># ?summary.survfit
range(lung$time)
seq(0, 1100, 100)</code></pre>
<p>And we can use that sequence vector with a summary call on sfit to get life tables at those intervals separately for both males (1) and females (2). From these tables we can start to see that males tend to have worse survival than females.</p>
<pre class="r"><code>summary(sfit, times=seq(0, 1000, 100))</code></pre>
<pre><code>## Call: survfit(formula = Surv(time, status) ~ sex, data = lung)
## 
##                 sex=1 
##  time n.risk n.event survival std.err lower 95% CI upper 95% CI
##     0    138       0   1.0000  0.0000       1.0000        1.000
##   100    114      24   0.8261  0.0323       0.7652        0.892
##   200     78      30   0.6073  0.0417       0.5309        0.695
##   300     49      20   0.4411  0.0439       0.3629        0.536
##   400     31      15   0.2977  0.0425       0.2250        0.394
##   500     20       7   0.2232  0.0402       0.1569        0.318
##   600     13       7   0.1451  0.0353       0.0900        0.234
##   700      8       5   0.0893  0.0293       0.0470        0.170
##   800      6       2   0.0670  0.0259       0.0314        0.143
##   900      2       2   0.0357  0.0216       0.0109        0.117
##  1000      2       0   0.0357  0.0216       0.0109        0.117
## 
##                 sex=2 
##  time n.risk n.event survival std.err lower 95% CI upper 95% CI
##     0     90       0   1.0000  0.0000       1.0000        1.000
##   100     82       7   0.9221  0.0283       0.8683        0.979
##   200     66      11   0.7946  0.0432       0.7142        0.884
##   300     43       9   0.6742  0.0523       0.5791        0.785
##   400     26      10   0.5089  0.0603       0.4035        0.642
##   500     21       5   0.4110  0.0626       0.3050        0.554
##   600     11       3   0.3433  0.0634       0.2390        0.493
##   700      8       3   0.2496  0.0652       0.1496        0.417
##   800      2       5   0.0832  0.0499       0.0257        0.270
##   900      1       0   0.0832  0.0499       0.0257        0.270</code></pre>
</div>
<div id="kaplan-meier-plots" class="section level3">
<h3>Kaplan-Meier Plots</h3>
<p>Now that we’ve fit a survival curve to the data it’s pretty easy to visualize it with a <strong>Kaplan-Meier</strong> plot. Create the survival object if you don’t have it yet, and instead of using <code>summary()</code>, use <code>plot()</code> instead.</p>
<pre class="r"><code>sfit &lt;- survfit(Surv(time, status)~sex, data=lung)
plot(sfit)</code></pre>
<p><img src="r-survival_files/figure-html/sfitPlot-1.png" width="672" /></p>
<p>There are lots of ways to modify the plot produced by base R’s <code>plot()</code> function. You can see more options with the help for <code>?plot.survfit</code>. We’re not going to go into any more detail here, because there’s another package called <strong>survminer</strong> that provides a function called <strong><code>ggsurvplot()</code></strong> that makes it much easier to produce publication-ready survival plots, and if you’re familiar with ggplot2 syntax it’s pretty easy to modify. So, let’s load the package and try it out.</p>
<pre class="r"><code>library(survminer)
ggsurvplot(sfit)</code></pre>
<p><img src="r-survival_files/figure-html/survminer-1.png" width="672" /></p>
<p>This plot is substantially more informative by default, just because it automatically color codes the different groups, adds axis labels, and creates and automatic legend. But there’s a lot more you can do pretty easily here. Let’s add confidence intervals, show the p-value for the log-rank test, show a risk table below the plot, and change the colors and the group labels.</p>
<pre class="r"><code>ggsurvplot(sfit, conf.int=TRUE, pval=TRUE, risk.table=TRUE, 
           legend.labs=c(&quot;Male&quot;, &quot;Female&quot;), legend.title=&quot;Sex&quot;,  
           palette=c(&quot;dodgerblue2&quot;, &quot;orchid2&quot;), 
           title=&quot;Kaplan-Meier Curve for Lung Cancer Survival&quot;, 
           risk.table.height=.15)</code></pre>
<p><img src="r-survival_files/figure-html/survminerOptions-1.png" width="672" /></p>
<hr />
</div>
<div id="exercise-set-1" class="section level3">
<h3>Exercise set 1</h3>
<p>Take a look at the built in <code>colon</code> dataset. If you type <code>?colon</code> it’ll ask you if you wanted help on the colon dataset from the survival package, or the colon operator. Click “Chemotherapy for Stage B/C colon cancer”, or be specific with <code>?survival::colon</code>. This dataset has survival and recurrence information on 929 people from a clinical trial on colon cancer chemotherapy. There are two rows per person, indidicated by the event type (<code>etype</code>) variable – <code>etype==1</code> indicates that row corresponds to recurrence; <code>etype==2</code> indicates death.</p>
<p>First, let’s turn the colon data into a tibble, then filter the data to only include the survival data, not the recurrence data. Let’s call this new object <code>colondeath</code>. The <code>filter()</code> function is in the <strong>dplyr</strong> library, which you can get by running <code>library(dplyr)</code>. If you don’t have dplyr you can use the base <code>subset()</code> function instead.</p>
<pre class="r"><code>library(dplyr)
colon &lt;- as_tibble(colon)
colondeath &lt;- filter(colon, etype==2)

# Or, using base subset()
# colondeath &lt;- subset(colon, etype==2)

head(colondeath)</code></pre>
<ol style="list-style-type: decimal">
<li><p>Look at the help for <code>?colon</code> again. How are <code>sex</code> and <code>status</code> coded? How is this different from the lung data?</p></li>
<li><p>Using <code>survfit(Surv(..., ...,)~..., data=colondeath)</code>, create a survival curve separately for males versus females. Call the resulting object <code>sfit</code>. Run a <code>summary()</code> on this object, showing time points 0, 500, 1000, 1500, and 2000. Do males or females appear to fair better over this time period?</p></li>
</ol>
<pre><code>##                 sex=0 
##  time n.risk n.event survival std.err lower 95% CI upper 95% CI
##     0    445       0    1.000  0.0000        1.000        1.000
##   500    381      64    0.856  0.0166        0.824        0.889
##  1000    306      75    0.688  0.0220        0.646        0.732
##  1500    265      40    0.598  0.0232        0.554        0.645
##  2000    218      22    0.547  0.0236        0.503        0.596
## 
##                 sex=1 
##  time n.risk n.event survival std.err lower 95% CI upper 95% CI
##     0    484       0    1.000  0.0000        1.000        1.000
##   500    418      65    0.866  0.0155        0.836        0.897
##  1000    335      83    0.694  0.0210        0.654        0.736
##  1500    287      46    0.598  0.0223        0.556        0.644
##  2000    238      25    0.545  0.0227        0.503        0.592</code></pre>
<ol start="3" style="list-style-type: decimal">
<li>Using the survminer package, plot a Kaplan-Meier curve for this analysis with confidence intervals and showing the p-value. See <code>?ggsurvplot</code> for help. Is there a significant difference between males and females?</li>
</ol>
<p><img src="r-survival_files/figure-html/unnamed-chunk-6-1.png" width="672" /></p>
<ol start="4" style="list-style-type: decimal">
<li>Create Kaplan-Meier plot stratifying by:
<ol style="list-style-type: lower-alpha">
<li>The extent of differentiation (well, moderate, poor), showing the p-value.</li>
<li>Whether or not there was detectable cancer in &gt;=4 lymph nodes, showing the p-value and confidence bands.</li>
</ol></li>
</ol>
<p><img src="r-survival_files/figure-html/unnamed-chunk-7-1.png" width="672" /><img src="r-survival_files/figure-html/unnamed-chunk-7-2.png" width="672" /></p>
<hr />
</div>
<div id="cox-regression" class="section level3">
<h3>Cox Regression</h3>
<p>Kaplan-Meier curves are good for visualizing differences in survival between two categorical groups, and the log-rank test you get when you ask for <code>pval=TRUE</code> is useful for asking if there are differences in survival between different groups. But this doesn’t generalize well for assessing the effect of <em>quantitative</em> variables. Just try creating a K-M plot for the <code>nodes</code> variable, which has values that range from 0-33. What a mess! Don’t do this.</p>
<pre class="r"><code>ggsurvplot(survfit(Surv(time, status)~nodes, data=colondeath))</code></pre>
<p><img src="r-survival_files/figure-html/unnamed-chunk-8-1.png" width="672" /></p>
<p>At some point using a categorical grouping for K-M plots breaks down, and further, you might want to assess how <em>multiple</em> variables work together to influence survival. For example, you might want to simultaneously examine the effect of race and socioeconomic status, so as to adjust for factors like income, access to care, etc., before concluding that ethnicity influences some outcome.</p>
<p>Cox PH regression can assess the effect of both categorical and continuous variables, and can model the effect of multiple variables at once. The <code>coxph()</code> function uses the same syntax as <code>lm()</code>, <code>glm()</code>, etc. The response variable you create with <code>Surv()</code> goes on the left hand side of the formula, specified with a <code>~</code>. Explanatory variables go on the right side.</p>
<p>Let’s go back to the lung cancer data and run a Cox regression on sex.</p>
<pre class="r"><code>fit &lt;- coxph(Surv(time, status)~sex, data=lung)
fit</code></pre>
<pre><code>## Call:
## coxph(formula = Surv(time, status) ~ sex, data = lung)
## 
##       coef exp(coef) se(coef)     z      p
## sex -0.531     0.588    0.167 -3.18 0.0015
## 
## Likelihood ratio test=10.6  on 1 df, p=0.00111
## n= 228, number of events= 165</code></pre>
<p>The <code>exp(coef)</code> column contains <span class="math inline">\(e^{\beta_1}\)</span> (see <a href="#background">background</a> section above for more info). This is the <strong>hazard ratio</strong> – the multiplicative effect of that variable on the hazard rate (for each unit increase in that variable). So, for a categorical variable like sex, going from male (baseline) to female results in approximately ~40% reduction in hazard. You could also flip the sign on the <code>coef</code> column, and take <code>exp(0.531)</code>, which you can interpret as being male resulting in a 1.7-fold increase in hazard, or that males die ad approximately 1.7x the rate per unit time as females (females die at 0.588x the rate per unit time as males).</p>
<p>Just remember:</p>
<ul>
<li>HR=1: No effect</li>
<li>HR&gt;1: Increase in hazard</li>
<li>HR&lt;1: Reduction in hazard (protective)</li>
</ul>
<p>You’ll also notice there’s a p-value on the <code>sex</code> term, and a p-value on the overall model. That 0.00111 p-value is really close to the p=0.00131 p-value we saw on the Kaplan-Meier plot. That’s because the KM plot is showing the log-rank test p-value. You can get this out of the Cox model with a call to <code>summary(fit)</code>. You can directly calculate the log-rank test p-value using <code>survdiff()</code>.</p>
<pre class="r"><code>summary(fit)
survdiff(Surv(time, status)~sex, data=lung)</code></pre>
<p>Let’s create another model where we analyze all the variables in the dataset! This shows us how all the variables, when considered together, act to influence survival. Some are very strong predictors (sex, ECOG score). Interestingly, the Karnofsky performance score as rated by the physician was marginally significant, while the same score as rated by the patient was not.</p>
<pre class="r"><code>fit &lt;- coxph(Surv(time, status)~sex+age+ph.ecog+ph.karno+pat.karno+meal.cal+wt.loss, data=lung)
fit</code></pre>
<pre><code>## Call:
## coxph(formula = Surv(time, status) ~ sex + age + ph.ecog + ph.karno + 
##     pat.karno + meal.cal + wt.loss, data = lung)
## 
##                coef exp(coef)  se(coef)     z      p
## sex       -5.51e-01  5.76e-01  2.01e-01 -2.74 0.0061
## age        1.06e-02  1.01e+00  1.16e-02  0.92 0.3591
## ph.ecog    7.34e-01  2.08e+00  2.23e-01  3.29 0.0010
## ph.karno   2.25e-02  1.02e+00  1.12e-02  2.00 0.0457
## pat.karno -1.24e-02  9.88e-01  8.05e-03 -1.54 0.1232
## meal.cal   3.33e-05  1.00e+00  2.60e-04  0.13 0.8979
## wt.loss   -1.43e-02  9.86e-01  7.77e-03 -1.84 0.0652
## 
## Likelihood ratio test=28.3  on 7 df, p=0.000192
## n= 168, number of events= 121 
##    (60 observations deleted due to missingness)</code></pre>
<hr />
</div>
<div id="exercise-set-2" class="section level3">
<h3>Exercise set 2</h3>
<p>Let’s go back to the <code>colon</code> cancer dataset. Remember, you created a <code>colondeath</code> object in the first exercise that only includes survival (<code>etype==2</code>), not recurrence data points. See <code>?colon</code> for more information about this dataset.</p>
<ol style="list-style-type: decimal">
<li>Take a look at <code>levels(colondeath$rx)</code>. This tells you that the <code>rx</code> variable is the type of treatment the patient was on, which is either nothing (coded <code>Obs</code>, short for Observation), Levamisole (coded <code>Lev</code>), or Levamisole + 5-fluorouracil (coded <code>Lev+5FU</code>). This is a factor variable coded with these levels, in that order. This means that <code>Obs</code> is treated as the baseline group, and other groups are dummy-coded to represent the respective group.</li>
</ol>
<table>
<caption>With <em>k</em> levels of a categorical factor variable, you get <em>k-1</em> dummy variables created, each 0/1, indicating that the sample is a particular non-reference category. Having value 0 for all dummy variables indicates that the sample is baseline.</caption>
<thead>
<tr class="header">
<th align="left">rx</th>
<th align="right">Lev</th>
<th align="right">Lev+5FU</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Obs</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">Lev</td>
<td align="right">1</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">Lev+5FU</td>
<td align="right">0</td>
<td align="right">1</td>
</tr>
</tbody>
</table>
<ol start="2" style="list-style-type: decimal">
<li>Run a Cox proportional hazards regression model against this <code>rx</code> variable. How do you interpret the result? Which treatment seems to be significantly different from the control (<code>Obs</code>ervation)?</li>
</ol>
<pre><code>##              coef exp(coef) se(coef)     z      p
## rxLev     -0.0266    0.9737   0.1103 -0.24 0.8092
## rxLev+5FU -0.3717    0.6896   0.1188 -3.13 0.0017
## 
## Likelihood ratio test=12.2  on 2 df, p=0.0023
## n= 929, number of events= 452</code></pre>
<ol start="3" style="list-style-type: decimal">
<li>Show the results using a Kaplan-Meier plot, with confidence intervals and the p-value.</li>
</ol>
<p><img src="r-survival_files/figure-html/unnamed-chunk-14-1.png" width="672" /></p>
<ol start="4" style="list-style-type: decimal">
<li>Fit another Cox regression model accounting for age, sex, and the number of nodes with detectable cancer. Notice the test statistic on the likelihood ratio test becomes much larger, and the overall model becomes more significant. What do you think accounted for this increase in our ability to model survival?</li>
</ol>
<pre><code>##               coef exp(coef) se(coef)     z       p
## rxLev     -0.08007   0.92305  0.11161 -0.72 0.47312
## rxLev+5FU -0.40253   0.66863  0.12054 -3.34 0.00084
## age        0.00533   1.00535  0.00405  1.32 0.18739
## sex       -0.02826   0.97214  0.09573 -0.30 0.76786
## nodes      0.09275   1.09719  0.00887 10.46 &lt; 2e-16
## 
## Likelihood ratio test=87.8  on 5 df, p=0
## n= 911, number of events= 441 
##    (18 observations deleted due to missingness)</code></pre>
<hr />
</div>
<div id="categorizing-for-km-plots" class="section level3">
<h3>Categorizing for KM plots</h3>
<p>Let’s go back to the lung data and look at a Cox model for age. Looks like age is very slightly significant when modeled as a continuous variable.</p>
<pre class="r"><code>coxph(Surv(time, status)~age, data=lung)</code></pre>
<p>Now that your regression analysis shows you that age is marginally significant, let’s make a Kaplan-Meier plot. But, as we saw before, we can’t just do this, because we’ll get a separate curve for every unique value of age!</p>
<pre class="r"><code>ggsurvplot(survfit(Surv(time, status)~age, data=lung))</code></pre>
<p>One thing you might see here is an attempt to categorize a continuous variable into different groups – tertiles, upper quartile vs lower quartile, a median split, etc – so you can make the KM plot. But, how you make that cut is meaningful! Check out the help for <code>?cut</code>. <code>cut()</code> takes a continuous variable and some breakpoints and creats a categorical variable from that. Let’s get the average age in the dataset, and plot a histogram showing the distribution of age.</p>
<pre class="r"><code>mean(lung$age)
hist(lung$age)
ggplot(lung, aes(age)) + geom_histogram(bins=20)</code></pre>
<p>Now, let’s try creating a categorical variable on <code>lung$age</code> with cut pounts at 0, 62 (the mean), and +Infinity (no upper limit). We could continue adding a <code>labels=</code> option here to label the groupings we create, for instance, as “young” and “old”. Finally, we could assign the result of this to a new object in the lung dataset.</p>
<pre class="r"><code>cut(lung$age, breaks=c(0, 62, Inf))
cut(lung$age, breaks=c(0, 62, Inf), labels=c(&quot;young&quot;, &quot;old&quot;))

# the base r way:
lung$agecat &lt;- cut(lung$age, breaks=c(0, 62, Inf), labels=c(&quot;young&quot;, &quot;old&quot;))

# or the dplyr way:
lung &lt;- lung %&gt;% 
  mutate(agecat=cut(age, breaks=c(0, 62, Inf), labels=c(&quot;young&quot;, &quot;old&quot;)))

head(lung)</code></pre>
<p>Now, what happens when we make a KM plot with this new categorization? It looks like there’s some differences in the curves between “old” and “young” patients, with older patients having slightly worse survival odds. But at p=.39, the difference in survival between those younger than 62 and older than 62 are not significant.</p>
<pre class="r"><code>ggsurvplot(survfit(Surv(time, status)~agecat, data=lung), pval=TRUE)</code></pre>
<p><img src="r-survival_files/figure-html/unnamed-chunk-20-1.png" width="672" /></p>
<p>But, what if we chose a different cut point, say, 70 years old, which is roughly the cutoff for the upper quartile of the age distribution (see <code>?quantile</code>). The result is now marginally significant!</p>
<pre class="r"><code># the base r way:
lung$agecat &lt;- cut(lung$age, breaks=c(0, 70, Inf), labels=c(&quot;young&quot;, &quot;old&quot;))

# or the dplyr way:
lung &lt;- lung %&gt;% 
  mutate(agecat=cut(age, breaks=c(0, 70, Inf), labels=c(&quot;young&quot;, &quot;old&quot;)))

# plot!
ggsurvplot(survfit(Surv(time, status)~agecat, data=lung), pval=TRUE)</code></pre>
<p><img src="r-survival_files/figure-html/unnamed-chunk-21-1.png" width="672" /></p>
<p>Remember, the Cox regression analyzes the continuous variable over the whole range of its distribution, where the log-rank test on the Kaplan-Meier plot can change depending on how you categorize your continuous variable. They’re answering a similar question in a different way: the regression model is asking, <em>“what is the effect of age on survival?”</em>, while the log-rank test and the KM plot is asking, <em>“are there differences in survival between those less than 70 and those greater than 70 years old?”</em>.</p>
<p><em>(New in survminer 0.2.4: the survminer package can now determine the optimal cutpoint for one or multiple continuous variables at once, using the <code>surv_cutpoint()</code> and <code>surv_categorize()</code> functions. Refer to <a href="http://www.sthda.com/english/wiki/survminer-0-2-4#determine-the-optimal-cutpoint-for-continuous-variables">this blog post</a> for more information.)</em></p>
</div>
</div>
<div id="tcga" class="section level2">
<h2>TCGA</h2>
<p>The Cancer Genome Atlas (TCGA) is a collaboration between the National Cancer Institute (NCI) and the National Human Genome Research Institute (NHGRI) that collected lots of clinical and genomic data across 33 cancer types. The entire TCGA dataset is over 2 petabytes worth of gene expression, CNV profiling, SNP genotyping, DNA methylation, miRNA profiling, exome sequencing, and other types of data. You can learn more about TCGA at <a href="https://cancergenome.nih.gov">cancergenome.nih.gov</a>. The data is now housed at the <a href="https://gdc-portal.nci.nih.gov/">Genomic Data Commons Portal</a>. There are lots of ways to access TCGA data without actually downloading and parsing through the data from GDC. We’ll cover more of these below. But first, let’s look at an R package that provides convenient, direct access to TCGA data.</p>
<div id="rtcga" class="section level3">
<h3>RTCGA</h3>
<p>The RTCGA package (<a href="http://bioconductor.org/packages/RTCGA">bioconductor.org/packages/RTCGA</a>) and all the associated data packages provide convenient access to clinical and genomic data in TCGA. Each of the data packages is a separate package, and must be installed (once) individually.</p>
<pre class="r"><code># Load the bioconductor installer. 
# Try http:// if https:// doesn&#39;t work.
source(&quot;https://bioconductor.org/biocLite.R&quot;)

# Install the main RTCGA package
biocLite(&quot;RTCGA&quot;)

# Install the clinical and mRNA gene expression data packages
biocLite(&quot;RTCGA.clinical&quot;)
biocLite(&quot;RTCGA.mRNA&quot;)</code></pre>
<p>Let’s load the RTCGA package, and use the <code>infoTCGA()</code> function to get some information about the kind of data available for each cancer type.</p>
<pre class="r"><code>library(RTCGA)
infoTCGA()</code></pre>
<div id="survival-analysis-with-rtcga-clinical-data" class="section level4">
<h4>Survival Analysis with RTCGA Clinical Data</h4>
<p>Next, let’s load the <code>RTCGA.clinical</code> package and get a little help about what’s available there.</p>
<pre class="r"><code>library(RTCGA.clinical)
?clinical</code></pre>
<p>This tells us all the clinical datasets available for each cancer type. If we just focus on breast cancer, look at how big the data is! There are 1098 rows by 3703 columns in this data alone. Let’s look at some of the variable names. <strong><em>Be careful with <code>View()</code> here</em></strong> – with so many columns, depending on which version of RStudio you have that may or may not have fixed this issue, Viewing a large dataset like this may lock up your RStudio.</p>
<pre class="r"><code>dim(BRCA.clinical)
names(BRCA.clinical)
# View(BRCA.clinical)</code></pre>
<p>We’re going to use the <code>survivalTCGA()</code> function from the RTCGA package to pull out survival information from the clinical data. It does this by looking at vital status (dead or alive) and creating a <code>times</code> variable that’s either the days to death or the days followed up before being censored. Look at the help for <code>?survivalTCGA</code> for more info. You give it a list of clinical datasets to pull from, and a character vector of variables to extract. Let’s look at breast cancer, ovarian cancer, and glioblastoma multiforme. Let’s just extract the cancer type (<code>admin.disease_code</code>).</p>
<pre class="r"><code># Create the clinical data
clin &lt;- survivalTCGA(BRCA.clinical, OV.clinical, GBM.clinical, 
                     extract.cols=&quot;admin.disease_code&quot;)
# Show the first few lines
head(clin)</code></pre>
<pre><code>##   times bcr_patient_barcode patient.vital_status admin.disease_code
## 1  3767        TCGA-3C-AAAU                    0               brca
## 2  3801        TCGA-3C-AALI                    0               brca
## 3  1228        TCGA-3C-AALJ                    0               brca
## 4  1217        TCGA-3C-AALK                    0               brca
## 5   158        TCGA-4H-AAAK                    0               brca
## 6  1477        TCGA-5L-AAT0                    0               brca</code></pre>
<pre class="r"><code># How many samples of each type?
table(clin$admin.disease_code)</code></pre>
<pre><code>## 
## brca  gbm   ov 
## 1098  595  576</code></pre>
<pre class="r"><code># Tabulate by outcome
xtabs(~admin.disease_code+patient.vital_status, data=clin) %&gt;% addmargins()</code></pre>
<pre><code>##                   patient.vital_status
## admin.disease_code    0    1  Sum
##               brca  994  104 1098
##               gbm   149  446  595
##               ov    279  297  576
##               Sum  1422  847 2269</code></pre>
<p>Now let’s run a Cox PH model against the disease code. By default it’s going to treat breast cancer as the baseline, because alphabetically it’s first. But you can reorder this if you want with <code>factor()</code>.</p>
<pre class="r"><code>coxph(Surv(times, patient.vital_status)~admin.disease_code, data=clin)</code></pre>
<pre><code>## Call:
## coxph(formula = Surv(times, patient.vital_status) ~ admin.disease_code, 
##     data = clin)
## 
##                         coef exp(coef) se(coef)    z      p
## admin.disease_codegbm  2.887    17.948    0.113 25.6 &lt;2e-16
## admin.disease_codeov   1.547     4.697    0.115 13.4 &lt;2e-16
## 
## Likelihood ratio test=904  on 2 df, p=0
## n= 2269, number of events= 847</code></pre>
<p>This tells us that compared to the baseline <code>brca</code> group, GBM patients have a ~18x increase in hazards, and ovarian cancer patients have ~5x worse survival. Let’s create a survival curve, visualize it with a Kaplan-Meier plot, and show a table for the first 5 years survival rates.</p>
<pre class="r"><code>sfit &lt;- survfit(Surv(times, patient.vital_status)~admin.disease_code, data=clin)
summary(sfit, times=seq(0,365*5,365))</code></pre>
<pre><code>## Call: survfit(formula = Surv(times, patient.vital_status) ~ admin.disease_code, 
##     data = clin)
## 
##                 admin.disease_code=brca 
##  time n.risk n.event survival std.err lower 95% CI upper 95% CI
##     0   1096       0    1.000 0.00000        1.000        1.000
##   365    588      13    0.981 0.00516        0.971        0.992
##   730    413      11    0.958 0.00851        0.942        0.975
##  1095    304      20    0.905 0.01413        0.878        0.933
##  1460    207       9    0.873 0.01719        0.840        0.908
##  1825    136      14    0.799 0.02474        0.752        0.849
## 
##                 admin.disease_code=gbm 
##  time n.risk n.event survival std.err lower 95% CI upper 95% CI
##     0    595       2   0.9966 0.00237       0.9920       1.0000
##   365    224     257   0.5110 0.02229       0.4692       0.5567
##   730     75     127   0.1998 0.01955       0.1649       0.2420
##  1095     39      31   0.1135 0.01617       0.0858       0.1500
##  1460     27       9   0.0854 0.01463       0.0610       0.1195
##  1825     12       9   0.0534 0.01259       0.0336       0.0847
## 
##                 admin.disease_code=ov 
##  time n.risk n.event survival std.err lower 95% CI upper 95% CI
##     0    576       0    1.000  0.0000        1.000        1.000
##   365    411      59    0.888  0.0139        0.861        0.915
##   730    314      55    0.761  0.0198        0.724        0.801
##  1095    210      59    0.602  0.0243        0.556        0.651
##  1460    133      49    0.451  0.0261        0.402        0.505
##  1825     78      39    0.310  0.0260        0.263        0.365</code></pre>
<pre class="r"><code>ggsurvplot(sfit, conf.int=TRUE, pval=TRUE)</code></pre>
<p><img src="r-survival_files/figure-html/unnamed-chunk-25-1.png" width="672" /></p>
</div>
<div id="gene-expression-data" class="section level4">
<h4>Gene Expression Data</h4>
<p>Let’s load the gene expression data.</p>
<pre class="r"><code>library(RTCGA.mRNA)
?mRNA</code></pre>
<p>Take a look at the size of the <code>BRCA.mRNA</code> dataset, show a few rows and columns.</p>
<pre class="r"><code>dim(BRCA.mRNA)
BRCA.mRNA[1:5, 1:5]</code></pre>
<blockquote>
<p><strong>Extra credit assignment</strong>: Take a look at the <a href="r-dplyr-yeast.html">advanced data manipulation</a> and <a href="r-tidy.html">tidy data</a> classes, and see if you can figure out how to join the gene expression data to the clinical data for any particular cancer type.</p>
</blockquote>
<pre class="r"><code># Take the mRNA data
BRCA.mRNA %&gt;% 
  # then make it a tibble (nice printing while debugging)
  as_tibble() %&gt;% 
  # then get just a few genes
  select(bcr_patient_barcode, PAX8, GATA3, ESR1) %&gt;% 
  # then trim the barcode (see head(clin), and ?substr)
  mutate(bcr_patient_barcode = substr(bcr_patient_barcode, 1, 12)) %&gt;% 
  # then join back to clinical data
  inner_join(clin, by=&quot;bcr_patient_barcode&quot;)</code></pre>
<p>Similar to how <code>survivalTCGA()</code> was a nice helper function to pull out survival information from multiple different clinical datasets, <code>expressionsTCGA()</code> can pull out specific gene expression measurements across different cancer types. See the help for <code>?expressionsTCGA</code>. Let’s pull out data for PAX8, GATA-3, and the estrogen receptor genes from breast, ovarian, and endometrial cancer, and plot the expression of each with a box plot.</p>
<pre class="r"><code>library(ggplot2)
expr &lt;- expressionsTCGA(BRCA.mRNA, OV.mRNA, UCEC.mRNA,
                        extract.cols = c(&quot;PAX8&quot;, &quot;GATA3&quot;, &quot;ESR1&quot;))
head(expr)</code></pre>
<pre><code>## # A tibble: 6 x 5
##            bcr_patient_barcode   dataset   PAX8 GATA3   ESR1
##                          &lt;chr&gt;     &lt;chr&gt;  &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt;
## 1 TCGA-A1-A0SD-01A-11R-A115-07 BRCA.mRNA -0.542  2.87  3.084
## 2 TCGA-A1-A0SE-01A-11R-A084-07 BRCA.mRNA -0.595  2.17  2.386
## 3 TCGA-A1-A0SH-01A-11R-A084-07 BRCA.mRNA  0.500  1.32  0.791
## 4 TCGA-A1-A0SJ-01A-11R-A084-07 BRCA.mRNA -0.589  1.84  2.495
## 5 TCGA-A1-A0SK-01A-12R-A084-07 BRCA.mRNA -0.965 -6.03 -4.861
## 6 TCGA-A1-A0SM-01A-11R-A084-07 BRCA.mRNA  0.573  1.80  2.797</code></pre>
<pre class="r"><code>table(expr$dataset)</code></pre>
<pre><code>## 
## BRCA.mRNA   OV.mRNA UCEC.mRNA 
##       590       561        54</code></pre>
<pre class="r"><code>ggplot(expr, aes(dataset, PAX8, fill=dataset)) + geom_boxplot()</code></pre>
<p><img src="r-survival_files/figure-html/geneExprBoxplots-1.png" width="672" /></p>
<pre class="r"><code>ggplot(expr, aes(dataset, GATA3, fill=dataset)) + geom_boxplot()</code></pre>
<p><img src="r-survival_files/figure-html/geneExprBoxplots-2.png" width="672" /></p>
<pre class="r"><code>ggplot(expr, aes(dataset, ESR1, fill=dataset)) + geom_boxplot()</code></pre>
<p><img src="r-survival_files/figure-html/geneExprBoxplots-3.png" width="672" /></p>
<pre class="r"><code>ggplot(expr, aes(dataset, ESR1, fill=dataset)) + geom_violin()</code></pre>
<p><img src="r-survival_files/figure-html/geneExprBoxplots-4.png" width="672" /></p>
<p>We could also use tidyr to do this all in one go.</p>
<pre class="r"><code>library(tidyr)
expr %&gt;% 
  as_tibble() %&gt;% 
  gather(gene, expression, PAX8, GATA3, ESR1) %&gt;% 
  ggplot(aes(dataset, expression, fill=dataset)) + 
    geom_boxplot() + 
    facet_wrap(~gene)</code></pre>
<p><img src="r-survival_files/figure-html/tidyGeneExprBoxplots-1.png" width="960" /></p>
</div>
</div>
<div id="exercise-set-3" class="section level3">
<h3>Exercise set 3</h3>
<p>The “KIPAN” cohort (in <code>KIPAN.clinical</code>) is the pan-kidney cohort, consisting of KICH (chromaphobe renal cell carcinoma), KIRC (renal clear cell carcinoma), and KIPR (papillary cell carcinoma). The <code>KIPAN.clinical</code> has <code>KICH.clinical</code>, <code>KIRC.clinical</code>, and <code>KIPR.clinical</code> all combined.</p>
<ol style="list-style-type: decimal">
<li>Using <code>survivalTCGA()</code>, create a new object called <code>clinkid</code> using the <code>KIPAN.clinical</code> cohort. For the columns to extract, get both the disease code and the patient’s gender (<code>extract.cols=c(&quot;admin.disease_code&quot;, &quot;patient.gender&quot;)</code>). The first few rows will look like this.</li>
</ol>
<pre><code>##   times bcr_patient_barcode patient.vital_status admin.disease_code
## 1  1158        TCGA-KL-8323                    1               kich
## 2  4311        TCGA-KL-8324                    0               kich
## 3   725        TCGA-KL-8325                    1               kich
## 4  3322        TCGA-KL-8326                    0               kich
## 5  3553        TCGA-KL-8327                    0               kich
## 6  3127        TCGA-KL-8328                    0               kich
##   patient.gender
## 1         female
## 2         female
## 3         female
## 4           male
## 5         female
## 6           male</code></pre>
<ol start="2" style="list-style-type: decimal">
<li>The <code>xtabs()</code> command will produce tables of counts for categorical variables. Here’s an example for how to use <code>xtabs()</code> for the built-in colon cancer dataset, which will tell you the number of samples split by sex and by treatment.</li>
</ol>
<pre class="r"><code>xtabs(~rx+sex, data=colon)</code></pre>
<pre><code>##          sex
## rx          0   1
##   Obs     298 332
##   Lev     266 354
##   Lev+5FU 326 282</code></pre>
<p>Use the same command to examine how many samples you have for each kidney sample type, separately by sex.</p>
<pre><code>##                   patient.gender
## admin.disease_code female male
##               kich     51   61
##               kirc    191  346
##               kirp     76  212</code></pre>
<ol start="3" style="list-style-type: decimal">
<li>Run a Cox PH regression on the cancer type and gender. What’s the effect of gender? Is it significant? How does survival differ by each type? Which has the worst prognosis?</li>
</ol>
<pre><code>##                           coef exp(coef) se(coef)     z       p
## admin.disease_codekirc  1.5929    4.9179   0.3450  4.62 3.9e-06
## admin.disease_codekirp  0.9962    2.7080   0.3807  2.62  0.0089
## patient.gendermale     -0.0628    0.9391   0.1484 -0.42  0.6721
## 
## Likelihood ratio test=39.4  on 3 df, p=1.4e-08
## n= 937, number of events= 203</code></pre>
<ol start="4" style="list-style-type: decimal">
<li>Create survival curves for each different subtype.
<ol style="list-style-type: lower-alpha">
<li>Produce a Kaplan-Meier plot.</li>
<li>Show survival tables each year for the first 5 years.</li>
</ol></li>
</ol>
<p><img src="r-survival_files/figure-html/unnamed-chunk-34-1.png" width="672" /></p>
<pre><code>## Call: survfit(formula = Surv(times, patient.vital_status) ~ admin.disease_code, 
##     data = clinkid)
## 
##                 admin.disease_code=kich 
##  time n.risk n.event survival std.err lower 95% CI upper 95% CI
##     0    111       0    1.000  0.0000        1.000        1.000
##   365     86       2    0.980  0.0144        0.952        1.000
##   730     72       2    0.954  0.0226        0.911        0.999
##  1095     54       3    0.910  0.0329        0.848        0.977
##  1460     44       1    0.893  0.0366        0.824        0.967
##  1825     38       1    0.871  0.0415        0.794        0.957
## 
##                 admin.disease_code=kirc 
##  time n.risk n.event survival std.err lower 95% CI upper 95% CI
##     0    536       0    1.000  0.0000        1.000        1.000
##   365    385      49    0.895  0.0142        0.868        0.924
##   730    313      32    0.816  0.0186        0.781        0.853
##  1095    250      26    0.744  0.0217        0.703        0.788
##  1460    181      20    0.678  0.0243        0.633        0.728
##  1825    112      16    0.606  0.0277        0.554        0.663
## 
##                 admin.disease_code=kirp 
##  time n.risk n.event survival std.err lower 95% CI upper 95% CI
##     0    288       0    1.000  0.0000        1.000        1.000
##   365    145      10    0.941  0.0182        0.906        0.977
##   730    100       8    0.877  0.0278        0.824        0.933
##  1095     67       2    0.853  0.0316        0.793        0.917
##  1460     54       3    0.810  0.0388        0.737        0.889
##  1825     36       5    0.727  0.0495        0.636        0.831</code></pre>
</div>
<div id="other-tcga-resources" class="section level3">
<h3>Other TCGA Resources</h3>
<p>RTCGA isn’t the only resource providing easy access to TCGA data. In fact, it isn’t even the only R/Bioconductor package. Take a look at some of the other resources shown below.</p>
<ul>
<li><strong><a href="https://bioconductor.org/packages/TCGAbiolinks">TCGAbiolinks</a></strong>: another R package that allows direct query and analysis from the NCI GDC.
<ul>
<li>R package: <a href="https://bioconductor.org/packages/TCGAbiolinks">bioconductor.org/packages/TCGAbiolinks</a></li>
<li>Paper: <em>Nucleic Acids Research</em> 2015 DOI: <a href="http://nar.oxfordjournals.org/content/44/8/e71">10.1093/nar/gkv1507</a>.</li>
</ul></li>
<li><strong><a href="http://www.cbioportal.org/">cBioPortal</a></strong>: <a href="http://www.cbioportal.org/">cbioportal.org</a>
<ul>
<li>Nice graphical user interface</li>
<li>Quick/easy summary info on patients, demographics, mutations, copy number alterations, etc.</li>
<li>Query individual genes, find coexpressed genes</li>
<li>Survival analysis against different subtypes, expression, CNAs, etc.</li>
</ul></li>
<li><strong><a href="http://www.oncolnc.org/">OncoLnc</a></strong>: <a href="http://www.oncolnc.org/">oncolnc.org</a>
<ul>
<li>Focus on survival analysis and RNA-seq data.</li>
<li>Simple query interface across all cancers for any mRNA, miRNA, or lncRNA gene (try SERPINA1)</li>
<li>Precomputed Cox PH regression for every gene, for every cancer</li>
<li>Kaplan-Meier plots produced on demand</li>
</ul></li>
<li><a href="http://ibl.mdanderson.org/tanric/_design/basic/index.html">TANRIC</a>: focus on noncoding RNA</li>
<li><a href="http://mexpress.be/">MEXPRESS</a>: focus on methylation and gene expression</li>
</ul>
<hr />
</div>
</div>
<div class="footnotes">
<hr />
<ol>
<li id="fn1"><p>In the medical world, we typically think of <em>survival analysis</em> literally – tracking time until death. But, it’s more general than that – survival analysis models time until an <em>event</em> occurs (<em>any</em> event). This might be death of a biological organism. But it could also be the time until a hardware failure in a mechanical system, time until recovery, time someone remains unemployed after losing a job, time until a ripe tomato is eaten by a grazing deer, time until someone falls asleep in a workshop, etc. <em>Survival analysis</em> also goes by <em>reliability theory</em> in engineering, <em>duration analysis</em> in economics, and <em>event history analysis</em> in sociology.<a href="#fnref1">↩</a></p></li>
<li id="fn2"><p>This describes the most common type of censoring – <em>right censoring</em>. <em>Left censoring</em> less commonly occurs when the “start” is unknown, such as when an initial diagnosis or exposure time is unknown.<a href="#fnref2">↩</a></p></li>
<li id="fn3"><p>And, following the definitions above, assumes that the <em>cumulative hazard</em> ratio between two groups remains constant over time.<a href="#fnref3">↩</a></p></li>
<li id="fn4"><p>And there’s a chi-square-like statistical test for these differences called the <a href="https://en.wikipedia.org/wiki/Log-rank_test">log-rank test</a> that compare the survival functions categorical groups.<a href="#fnref4">↩</a></p></li>
<li id="fn5"><p>See the <a href="r-stats.html#multiple_regression">multiple regression</a> section of the essential statistics lesson.<a href="#fnref5">↩</a></p></li>
<li id="fn6"><p>Cox regression and the logrank test from <code>survdiff</code> are going to give you similar results most of the time. The log-rank test is asking if survival curves differ significantly between two groups. Cox regression is asking which of many categorical or continuous variables significantly affect survival.<a href="#fnref6">↩</a></p></li>
<li id="fn7"><p><code>Surv()</code> can also take start and stop times, to account for left censoring. See the help for <code>?Surv</code>.<a href="#fnref7">↩</a></p></li>
<li id="fn8"><p>Loprinzi et al. Prospective evaluation of prognostic variables from patient-completed questionnaires. North Central Cancer Treatment Group. <em>Journal of Clinical Oncology</em>. 12(3):601-7, 1994.<a href="#fnref8">↩</a></p></li>
<li id="fn9"><p>Where “dead” really refers to the occurance of the event (any event), not necessarily death.<a href="#fnref9">↩</a></p></li>
</ol>
</div>

<div class="footer">
This work is licensed under the  <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/" target="_blank">CC BY-NC-SA 4.0 Creative Commons License</a>.<br>
For more information, visit <a href="http://data.hsl.virginia.edu/" target="_blank">data.hsl.virginia.edu</a>.<br>
<a href="https://twitter.com/strnr" target="_blank"><i class="fa fa-twitter fa-lg"></i></a>&nbsp;
<a href="https://github.com/bioconnector/workshops" target="_blank"><i class="fa fa-github fa-lg"></i></a>&nbsp;
<!--
<a href="http://www.gettinggeneticsdone.com/" target="_blank"><i class="fa fa-rss fa-lg"></i></a>&nbsp;
-->
</div>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
